assets/js/base/components/title/index.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+/**
+ * Component that renders a block title.
+ */
+const Title = ( { children, className, level, ...props } ) => {
```

Comment:
Yeah... the logic is similar, but I thought the `<Title>` as a simple wrapper that adds the `wc-block-component__title` class name and the associated styles. So from my point of view, all other titles that have different styles should not use this component. In fact, I just created this component in order to avoid having a generic `wc-block__title` class name that was not associated with any component. Does that make sense?

However, in 401725a6df4b747f3cb540d4e70ac83d21a481f0 I renamed the `level` prop to `headingLevel` to keep it consistent between both blocks.

Reply:
So what's going through my mind here is that we could use the `Title` component for things we want to consistently react to any changes in "base" styles found across components implementing it. This would have more implications as we roll out the transition to Global Styles. 

From your comment, I'm guessing here that you think that `ProductTitle` is distinct enough, that we _don't_ want it tied into this potential style tree?

Reply:
Heading native elements (`h1`, `h2`...) already react to "base" styles by default. That happens with the current styling system and will also happen with global styles because themes will define styles for those elements. So I don't think using `Title` instead of `h2` makes any difference in that sense.

In some cases, we want to inherit some values from the base styles while "locking" some others. With the Cart block titles, for example, we want to inherit the heading color and heading font family, but we want to lock the font size to something smaller so it works with the Cart layout. With the product title, instead, we want the regular text font size & family and the link color.

At least, that's how it currently works, if you go to the Customizer in Storefront and set the Heading color to `#ff0000` and the link color to `#00ff00`, the _Cart_ block headings get the headings color:
![imatge](https://user-images.githubusercontent.com/3616980/82691451-36e41e80-9c5e-11ea-9439-a6083db86999.png)

While the product titles in the _All Products_ block get the link color:
![imatge](https://user-images.githubusercontent.com/3616980/82691426-2b90f300-9c5e-11ea-8924-b635d6673ee9.png)

Relating this with the `Title` component, I think it only makes sense to reuse the same component (with same class names & styles), if the values we want to inherit + the values we want to lock are the same. In this specific case, we want the Product title and the Cart title to look quite different, so in my opinion we don't get any advantage on using the same component for them.

Reply:
Thanks for explaining that @Aljullu!  I guess I was thinking since `Title` allowed for receiving classNames, that I was thinking css for the cart/checkout titles would inherit from `wc-block-component__title` and then override for their specific styles. But what you're saying is that by default `h2,3...` elements _already_ have styles inherited from the theme and there is no real overlap (nor will there ever be) with styles that the cart and checkout block might inherit from `wc-block-component__title` right? If so, then I think that makes sense.

Reply:
> Thanks for explaining that @Aljullu!  I guess I was thinking since `Title` allowed for receiving classNames, that I was thinking css for the cart/checkout titles would inherit from `wc-block-component__title` and then override for their specific styles.

That's a good point. In the future, if we want to use this component in other blocks we might need to add a `className` prop. For now, I don't think we need it yet.

> But what you're saying is that by default `h2,3...` elements _already_ have styles inherited from the theme and there is no real overlap (nor will there ever be) with styles that the cart and checkout block might inherit from `wc-block-component__title` right?

Yes, exactly.